spec/controllers/foods_controller_spec.rb

```diff
@@ -1,159 +1,191 @@
-# Message expectations are only used when there is no simpler way to specify
-# that an instance is receiving a specific message.
+require "rails_helper"
 
 RSpec.describe FoodsController, type: :controller do
-
-  # This should return the minimal set of attributes required to create a valid
-  # Food. As you add validations to Food, be sure to
-  # adjust the attributes here as well.
-  let(:valid_attributes) {
-    skip("Add a hash of attributes valid for your model")
-  }
-
-  let(:invalid_attributes) {
-    skip("Add a hash of attributes invalid for your model")
-  }
-
-  # This should return the minimal set of values that should be in the session
-  # in order to pass any filters (e.g. authentication) defined in
-  # FoodsController. Be sure to keep this updated too.
-  let(:valid_session) { {} }
-
-  describe "GET #index" do
-    it "assigns all foods as @foods" do
-      food = Food.create! valid_attributes
-      get :index, {}, valid_session
-      expect(assigns(:foods)).to eq([food])
-    end
+  let(:valid_attributes) do
+    FactoryGirl.attributes_for(:food)
   end
 
-  describe "GET #show" do
-    it "assigns the requested food as @food" do
-      food = Food.create! valid_attributes
-      get :show, {:id => food.to_param}, valid_session
-      expect(assigns(:food)).to eq(food)
+  context "not logged in" do
+    describe 'GET #index' do
+      it "should redirect to login" do
+        expect(get(:index)).to redirect_to(new_user_session_url)
+      end
     end
-  end
 
-  describe "GET #new" do
-    it "assigns a new food as @food" do
-      get :new, {}, valid_session
-      expect(assigns(:food)).to be_a_new(Food)
+    describe 'GET #new' do
+      it "should redirect to login" do
+        expect(get(:new)).to redirect_to(new_user_session_url)
+      end
     end
-  end
 
-  describe "GET #edit" do
-    it "assigns the requested food as @food" do
-      food = Food.create! valid_attributes
-      get :edit, {:id => food.to_param}, valid_session
-      expect(assigns(:food)).to eq(food)
+    describe 'GET #edit' do
+      it "should redirect to login" do
+        food = FactoryGirl.create(:food)
+        expect(get(:edit, id: food.to_param)).
+          to redirect_to(new_user_session_url)
+      end
     end
-  end
 
-  describe "POST #create" do
-    context "with valid params" do
-      it "creates a new Food" do
-        expect {
-          post :create, {:food => valid_attributes}, valid_session
-        }.to change(Food, :count).by(1)
+    describe 'POST #create' do
+      it "should redirect to login" do
+        expect(post(:create, valid_attributes)).
+          to redirect_to(new_user_session_url)
       end
+    end
 
-      it "assigns a newly created food as @food" do
-        post :create, {:food => valid_attributes}, valid_session
-        expect(assigns(:food)).to be_a(Food)
-        expect(assigns(:food)).to be_persisted
+    describe 'PATCH #update' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.